app/services/licences/process-supplementary-billing-flag.service.js

```diff
@@ -0,0 +1,78 @@
+'use strict'
+
+/**
+ * Orchestrates flagging a licence for supplementary billing
+ * @module ProcessSupplementaryBillingFlagService
+ */
+
+const ChargeVersionModel = require('../../models/charge-version.model.js')
+const FlagSupplementaryBillingService = require('./flag-supplementary-billing.service.js')
+const DetermineSupplementaryBillingYearsService = require('./determine-supplementary-billing-years.service.js')
+
+/**
+ * Orchestrates flagging a licence for supplementary billing based on the provided charge version.
+ *
+ * If a `chargeVersionId` is present in the payload, the service fetches the charge version, along with its relevant
+ * charge references and licence details.
+ *
+ * If the charge version has any two-part tariff indicators on any of its charge references then it calls the
+ * `DetermineSupplementaryBillingYearsService` where it will work out which financial year ends have been affected by
+ * the change in the charge version.
+ *
+ * If there are years that have been affected, these are then passed to our `FlagSupplementaryBillingService`, which
+ * checks if a bill run has already been sent for that affected year, and if it has it will then persist the flag in the
+ * `LicenceSupplementaryYear` table.
+ *
+ * @param {Object} payload - The payload from the request to be validated
+ */
+async function go (payload) {
+  try {
+    if (!payload.chargeVersionId) {
+      return
+    }
+
+    const { chargeReferences, licence, endDate, startDate } = await _fetchChargeVersion(payload.chargeVersionId)
+    const twoPartTariffIndicator = _twoPartTariffIndicators(chargeReferences)
+
+    if (!twoPartTariffIndicator) {
+      return
+    }
+
+    const years = DetermineSupplementaryBillingYearsService.go(startDate, endDate)
+
+    if (years) {
+      await FlagSupplementaryBillingService.go(licence, years)
+    }
+  } catch (error) {
+    global.GlobalNotifier.omfg('Supplementary Billing Flag failed', error)
```

Comment:
This will break the notifier. `omfg()` expects 3 args. error is the last one, so if we want to pass it through we _must_ provide a second arg for data (else error is treated as the data arg by the function).

What I would actually suggest is you pass the `payload` as the second arg. You were right in that we couldn't use whatever the fetch returned because it might be the fetch erroring. But the payload is always available and having the details of _what_ triggered the call would be extremely useful when trying to resolve any future error/issue.